--- a/ide/app/spark.dart.precompiled.js
+++ b/ide/app/spark.dart.precompiled.js
@@ -31736,7 +31736,7 @@ _wrapToDart: function(o) {
 
 _getDartProxy: function(o, propertyName, createProxy) {
   var dartProxy = o[propertyName];
-  if (dartProxy == null) {
+  if (dartProxy == null || (!(o instanceof Object))) {
     dartProxy = createProxy.call$1(o);
     P._defineProperty(o, propertyName, dartProxy);
   }
